packages/affiliates/libs/uniswap/models.ts

```diff
@@ -0,0 +1,253 @@
+import assert from "assert";
+import { exists, getPositionKey } from "./utils";
+import { ethers } from "ethers";
+
+type CacheType = {
+  [key: string]: string | number | boolean;
+};
+export const Cache = (table = new Map<string | number, CacheType>()) => {
+  function list() {
+    return [...table.values()];
+  }
+  return {
+    has: table.has.bind(table),
+    set: table.set.bind(table),
+    get: table.get.bind(table),
+    delete: table.delete.bind(table),
+    list
+  };
+};
+export type NftPosition = {
+  id?: string;
+  tokenId: string;
+  amount?: string;
+  amount0?: string;
+  amount1?: string;
+  token0?: string;
+  token1?: string;
+  liquidity?: string;
+  feeGrowthInside0LastX128?: string;
+  feeGrowthInside1LastX128?: string;
+  tokensOwed0?: string;
+  tokensOwed1?: string;
+  pool?: string;
+};
+
+// Nft positions are gathered from the nft contract, and require slightly different indexing than the positions in pool contract
+export const NftPositions = (config: any, table: ReturnType<typeof Cache>) => {
+  function makeId(data: NftPosition) {
+    return data.tokenId;
```

Comment:
Why do you have both an `id` and `tokenId` in the `NftPosition` type? `tokenId` is required and is used to construct a new `NftPosition`, so in this case `id` seems redundant?

Reply:
you are right, i was keeping it as "id" more as a convention to say you can always trust .id to be the canonical way to lookup any model, where as tokenId would require context about the data.